apps/vor/pkg/postgres/classStore.go

```diff
@@ -26,7 +26,47 @@ func (s ClassStore) CheckPermission(userId string, classId string) (bool, error)
 	}
 	return true, nil
 }
+func (s ClassStore) GetClassSession(classId string) ([]class.ClassSession, error) {
 
+	var attendance []Attendance
+	var session []class.ClassSession
+	if err := s.DB.Model(&attendance).
+		Distinct().
+		ColumnExpr("cast(date AS date)").
+		Select(); err != nil {
+		return nil, err
+	}
+	if len(attendance) > 0 {
+		for _, attendance := range attendance {
+			session = append(session, class.ClassSession{
+				Date: attendance.Date.Format("2006-01-02"),
+			})
+		}
+	}
+	var selectedClass Class
+	if err := s.DB.Model(&selectedClass).
+		Relation("Weekdays").
+		Where("id=?", classId).
+		Select(); err != nil {
+		return nil, err
+	}
+	for _, weekday := range selectedClass.Weekdays {
+		if int(time.Weekday(weekday.Day)-time.Now().Weekday()) < 0 {
```

Comment:
is it better if we put this in a variable?

Reply:
which variable?

Reply:
I think she meant `time.Weekday(weekday.Day)-time.Now().Weekday()`  ?